idunn/blocks/opening_hour.py

```diff
@@ -66,30 +61,55 @@ def from_es(cls, es_poi, lang):
         raw = es_poi.get('properties', {}).get('opening_hours')
         if raw is None:
             return None
+
+        poi_coord = get_coord(es_poi)
+        poi_lat = poi_coord[0]
+        poi_lon = poi_coord[1]
+
         is247 = raw == '24/7'
 
-        try:
-            oh = hoh.OHParser(raw)
-        except HOHError:
-            logger.info("Failed to parse OSM opening_hour field", exc_info=True)
-            return None
+        poi_tzname = tz.tzNameAt(poi_lat, poi_lon, forceTZ=True)
+        poi_tz = get_tz(poi_tzname, poi_lat, poi_lon)
+        poi_location = (poi_lat, poi_lon, poi_tzname, 24)
 
-        poi_tz = get_tz(es_poi)
         if poi_tz is None:
             logger.info("No timezone found for poi %s", es_poi.get('id'))
             return None
 
+        try:
+            oh = hoh.OHParser(raw, location=poi_location)
+        except HOHError:
+            logger.info("Failed to parse OSM opening_hour field", exc_info=True)
+            return None
+
         poi_dt = UTC.localize(datetime.utcnow()).astimezone(poi_tz)
+
+        if oh.is_open(poi_dt.replace(tzinfo=None)):
+            status = 'open'
+        else:
+            status = 'closed'
+
+        if is247:
+            return cls(
+                status=status,
+                next_transition_datetime=None,
+                seconds_before_next_transition=None,
+                is_24_7=is247,
+                raw=oh.field,
+                days=cls.get_days(oh, dt=poi_dt)
+            )
+
         # The current version of the hoh lib doesn't allow to use the next_change() function
         # with an offset aware datetime.
         # This is why we replace the timezone info until this problem is fixed in the library.
         try:
             nt = oh.next_change(dt=poi_dt.replace(tzinfo=None))
+        except NextChangeRecursionError:
+            logger.info("NextChangeRecursionError: Failed to compute next transition for poi %s", es_poi.get('id'), exc_info=True)
```

Comment:
Is not that exception already handled by `HOHError` ?  
Actually the root cause of the 500 error is not fixed: we need to return when the exception is catched (otherwise `nt` is not defined)